test/space_test.jl

```diff
@@ -152,4 +152,23 @@ end
   kill_agent!(agent, model)
   @test_throws KeyError id2agent(1, model) 
   @test !in(1, Agents.agent_positions(model)[agent_pos])
-end
\ No newline at end of file
+end
+
+@testset "nodes" begin
+  mutable struct MyAgent <: AbstractAgent
+    id::Int
+    pos::Tuple{Int,Int}
+  end
+  space = Space((3,3))
+  model = ABM(MyAgent, space)
+  for node in nodes(model)
+    if rand() > 0.7
+      add_agent!(node, model)
+    end
+  end
+  @test nodes(model) == 1:9
+  @test nodes(model, by=:random) == [8, 9, 1, 7, 6, 5, 3, 2, 4]
```

Comment:
Shouldn't you set the seed here, to ensure that this random sequence will be the same all the time?

Reply:
There's a seed set at the beginning of the file.